src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,120 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int MAXIMUM_CAPACITY = Integer.MAX_VALUE / 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int VALUE_FOR_INCREASE = 2;
+    private Node<K, V>[] entryTable;
+    private int size;
+    private int threshold;
+    private final float loadFactor;
+
+    public MyHashMap() {
+        this(DEFAULT_INITIAL_CAPACITY, DEFAULT_LOAD_FACTOR);
+    }
+
+    private MyHashMap(int initialCapacity, float loadFactor) {
+        if (initialCapacity < 0) {
+            throw new IllegalArgumentException("Wrong initial capacity: " + initialCapacity);
+        }
+
+        if (initialCapacity > MAXIMUM_CAPACITY) {
+            initialCapacity = MAXIMUM_CAPACITY;
+        }
+
+        if (loadFactor <= 0 || Float.isNaN(loadFactor)) {
+            throw new IllegalArgumentException("Wrong load factor: " + loadFactor);
+        }
+
+        int capacity = 1;
+        while (capacity < initialCapacity) {
+            capacity++;
+        }
+
+        entryTable = new Node[capacity];
+        this.loadFactor = loadFactor;
+        threshold = (int) (capacity * loadFactor);
+    }
+
     @Override
     public void put(K key, V value) {
+        int numberOfBucket = getBucket(key);
+        Node<K, V> element;
+        if (size >= threshold) {
+            changeSize(entryTable.length * VALUE_FOR_INCREASE);
+        }
+        for (element = entryTable[numberOfBucket]; element != null; element = element.next) {
+            if (numberOfBucket == element.hash && Objects.equals(key, element.nodeKey)) {
+                element.nodeValue = value;
+                return;
+            } else if (numberOfBucket == element.hash && element.next == null) {
+                element.next = new Node<>(numberOfBucket, key, value, null);
+                size++;
+                return;
+            }
```

Comment:
In my opinion this else block is redudant, you're doing the same action after your cycle

Reply:
This is a part of the realization of the put() method. (I need a second check)